File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week2/code/worksheet/map_exercise.java
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 25+

Map<Integer, List<Integer>> build_less_than_map(List<Integer> lst) {
	var ret = new HashMap<Integer, List<Integer>>();

	for (Integer key : lst) {
		List<Integer> valLst = ret.computeIfAbsent(key, ArrayList<Integer>::new);

		for (Integer val : lst) {
			if (key > val && !valLst.contains(val)) {
				valLst.add(val);
			}
		}
	}

	return ret;
}

void main(String... args) {
	var ret = build_less_than_map(List.of(4, 1, 3, 3));
	System.out.println(ret);
}

File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week2/code/worksheet/map_exercise.py
def build_less_than_map(L):
    result = {}
    for x in L:
        if x not in result:
            result[x] = []
        for y in L:
            if y < x:
                if y not in result[x]:
                    result[x].append(y)
    return result

L = [4, 1, 3, 3]
m = build_less_than_map(L)
print(m)
